javalib/src/main/scala/java/util/PriorityQueue.scala

```diff
@@ -227,18 +279,18 @@ class PriorityQueue[E] private (
     /* At each step, even though `m` changes, the element moves with it, and
      * hence inner(m) is always the same initial `innerAtM`.
      */
-    val innerAtM = inner(m)
+    val innerAtM = innerImpl.get(inner, m)
 
     @inline @tailrec
     def loop(m: Int): Unit = {
       if (m > 1) {
         val parent = m >> 1
-        val innerAtParent = inner(parent)
+        val innerAtParent = innerImpl.get(inner, parent)
         if (comp.compare(innerAtParent, innerAtM) > 0) {
-          inner(parent) = innerAtM
-          inner(m) = innerAtParent
-          loop(parent)
+          innerImpl.set(inner, parent, innerAtM)
+          innerImpl.set(inner, m, innerAtParent)
         }
+        loop(parent)
```

Comment:
This is where the `loop(parent)` was moved to the wrong place. So the "fix `fixUp`" commit should be squashed with this one. It was correct in the original code. ;)